app/models/auth_token.rb

```diff
@@ -3,10 +3,7 @@
 class AuthToken < ActiveRecord::Base
   belongs_to :access_code
 
-  validates :access_code, presence: true
```

Comment:
While you may have DB constraints, I typically prefer to lean on application level validation. DB constraints is obfuscated rules. Having both makes it clear what will happen when you try to save.